Makefile

```diff
@@ -382,6 +402,10 @@ else
 	endif
 endif
 
+MPI_SRC = $(wildcard  src/mpi_collectives/src/*.cc)
```

Comment:
SRC_FILTER is the same.

Reply:
@szha I know that. SRC_FILTER is put here for future extension. Because maybe in future, another source code which is put within src/ need different compilation rule same as MPI_SRC.  SRC_FILTER will looks like  SRC_FILTER += $(wildcard src/<other-component>/src/*.cc)

Reply:
Let's put the simplest possible code that does the job in the codebase, and let the next person worry about how to use different compilation rules for new files. Most likely if anyone other than you works on this code, your comment here will be no where to be found by the contributor. "Potential extension" is often a synonyms for tech debt.

Reply:
OK. Then how about:
MPI_SRC = $(wildcard  src/mpi_collectives/src/*.cc)
...
SRC_FILTER = $(MPI_SRC)